### ISSUE TITLE
Mac version doesn't change bed temperature after the first layer

### ISSUE THREAD

**Issue by thonghh** [alexrj/Slic3r#3505]
### Version
1.2.9 Mac

### Operating system type + version
Mac OSX 10.10.5

### Behavior
* The problem is that I set the bed temperature to 40 degree for the first layer, 0 degree to the other layers in Filament Settings -> Filaments. But the printer doesn't perform what I expected. The bed temp is kept at 40 during the whole printing process.
* Set different bed temps in Filament Settings -> Filaments, but it doesn't change the temp at all

![screen shot 2016-09-16 at 5 40 40 pm](https://cloud.githubusercontent.com/assets/20840853/18583433/baefc060-7c34-11e6-9f82-9b8d59009ba6.png)

_Is this a new feature request?_

#### STL/Config (.ZIP) where problem occurs
_Upload a zipped copy of an STL and your config (`File -> Export Config`)_

**Comment by lordofhyphens** [comment]
Could you post a zipped copy of output gcode where this problem occurs? Use something simple like a 10x10 solid box.

**Comment by phizz166** [comment]
These values might be getting overwritten by your start gcode. Check that?

**Comment by thonghh** [comment]
I just sliced this gcode file, and the problem still occurs. Will check if setting to 10 degree after the first layer solves the problem.

[printer.zip](https://github.com/alexrj/Slic3r/files/486391/printer.zip)

**Comment by Pointedstick** [comment]
I'm seeing the same issue on OS X 12.10 with Slic3r 1.2.9, irrespective of whether or not I have any commands to set temperatures in my start.gcode file. The bed starts out heated at the first layer bed temperature, but does not change the temperature after the first layer. 100% reproducible for me.

**Comment by Pointedstick** [comment]
Update: this feature works properly when changing the temperature of subsequent layers to anything other than 0. But when you tell it to set the temperature of subsequent layers to zero, it ignores that.

**Comment by lordofhyphens** [comment]
Probably because 0 is interpreted as no value. Frankly anything less than
ambient  temp is effectively 0

**Comment by Pointedstick** [comment]
Right, I can set it to 10c and that works. But it's kinda weird and unintuitive.

**Comment by alexrj** [comment]
The tooltip for those options says: *Set this to zero to disable bed temperature control commands in the output.*.

**Comment by alexrj** [comment]
So, we have a dedicated "Has heatbed" option now which determines whether bed temperature commands are written.
However that new option is enabled by default, so if we change the semantics of first_layer_bed_temperature and bed_temperature to consider 0 as a valid value, all current users will have `M190 S0` and `M140 S0` in their output which is bad.
I just relaxed it only for bed_temperature. While it's not totally clean, I think it's a reasonable compromise.

Any thoughts?

**Comment by Pointedstick** [comment]
That sounds reasonable to me.
